api/envoy/config/bootstrap/v3/bootstrap.proto

```diff
@@ -305,10 +305,34 @@ message ClusterManager {
 // Envoy process watchdog configuration. When configured, this monitors for
 // nonresponsive threads and kills the process after the configured thresholds.
 // See the :ref:`watchdog documentation <operations_performance_watchdog>` for more information.
-// [#next-free-field: 7]
+// [#next-free-field: 8]
 message Watchdog {
   option (udpa.annotations.versioning).previous_message_type = "envoy.config.bootstrap.v2.Watchdog";
 
+  message WatchdogAction {
+    // In priority events are fired from KILL, MULTIKILL, MEGAMISS and MISS.
+    // Within an event type, actions execute in the order they are configured.
+    // For KILL/MULTIKILL there is a default PANIC that will kill the process,
+    // but it might be useful to specify several debug actions, and possibly
+    // an alternate FATAL action.
+    enum WatchdogEvent {
+      UNKNOWN = 0;
+      KILL = 1;
+      MULTIKILL = 2;
+      MEGAMISS = 3;
+      MISS = 4;
+    }
+
+    // Extension specific configuration for the action.
+    core.v3.TypedExtensionConfig config = 1;
```

Comment:
We have been really bad about keeping https://www.envoyproxy.io/docs/envoy/latest/extending/extending up to date, but can you add this there? Bonus points for searching for TypedExtensionConfig and seeing if there is anything else missing from that page. Ref linking to the config field would be great.

Reply:
Added the field to that doc and updated other TypedExtensionConfigs with references based on what I could tell from PRs.